dali/pipeline/operators/detection/box_encoder.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+
+#include "dali/pipeline/operators/detection/box_encoder.h"
+
+namespace dali {
+vector<float> BoxEncoder<CPUBackend>::CalculateIous(vector<BoundingBox> boxes) {
+  vector<float> ious(boxes.size() * NumAnchors());
+
+  for (auto bbox_idx = 0u; bbox_idx < boxes.size(); ++bbox_idx) {
+    int best_idx = -1;
+    float best_iou = -1.;
+
+    for (int anchor_idx = 0; anchor_idx < NumAnchors(); ++anchor_idx) {
+      ious[bbox_idx * anchors_.size() + anchor_idx] =
+          boxes[bbox_idx].IntersectionOverUnion(anchors_[anchor_idx]);
+
+      if (ious[bbox_idx * anchors_.size() + anchor_idx] >= best_iou) {
```

Comment:
You can declare bbox_idx * anchors_.size() + anchor_idx as you seem to be repeating it a little bit

Reply:
Fixed by extracting computation for each bounding box to separate method.